### DISCUSSION TITLE
How to disable ChromiumWebBrowser  remember username and password information on some authorization page.

### DISCUSSION THREAD

**Discussion by LuxWang** [cefsharp/CefSharp#3512] [General]
We use OidcClient.LoginAsync() and CefSharp.Wpf.ChromiumWebBrowser to load a webPage to do some authorization.
(for example: 
1. the webpage for authorization: https://signin.XXX.net 
2. we could enter the username, password to login
3. if the user info is correct it would redirect to another https://account.XXX.net/)

However, we found ChromiumWebBrowser may remember this username and password information. 
1. If we close the ChromiumWebBrowser window.
2. Later we launch another ChromiumWebBrowser window and connect to "https://signin.XXX.net", 
it would automatically redirect to the "https://account.XXX.net/".
(We guess some username and password information is automatically filled in "https://signin.XXX.net" webpage.)

We also can reproduce it in CefSharp.Wpf.example.
the step to reproduce:
1) go to "https://account.XXX.net/" page 
2) enter the correct username and password.
3) it would redirect to "https://account.XXX.net/" and show your account information.
4) close this tab.
5) open another tab, go to "https://account.XXX.net/" page 
6) It would automatically redirect to "https://account.XXX.net/". (The user doesn't need to enter username and password information.)

However, we would like the user to enter sign-in information every time. (with the same username or password or with a different username or password.)
Do you have any way to clear this user information (maybe in cache or user session or someplace)??
Could we set some property in CefSettings or ChromiumWebBrowser to achieve it?

Thank you so much for your help.

**Comment by amaitland** [comment]
The behaviour you are describing sounds pretty much as expected as a default.

> However, we would like the user to enter sign-in information every time. (maybe different username or password.)
> Do you have any way to clear this user information (maybe in cache or user session or someplace)??

How is the user information stored? Cookie?

If you need to just clear the login token for that website then you need to workout how it is stored.

If you wish to have a completely different cache for each `ChromiumWebBrowser` instance then you just need a unique `RequestContext` for each, see https://github.com/cefsharp/CefSharp/wiki/General-Usage#request-context-browser-isolation

**Reply by LuxWang** [reply]
Thank you very much for your help.

We can achieve the desired function by the following code.

//Default implementation of RequestContext
//Default settings will be used, this means an in-memory cache (no data persisted)
browser = new ChromiumWebBrowser();
browser.RequestContext = new RequestContext();
